src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -310,9 +293,39 @@ protected Object readResolve() {
         projectUpdateLock = new ReentrantLock();
         issueCache = makeIssueCache();
         jiraSession = null;
+
+        // HACK, update final fields via reflection, see jls 17.5.3, https://docs.oracle.com/javase/specs/jls/se7/html/jls-17.html#jls-17.5.3
+        if (StringUtils.isNotBlank(credentialsId)) {
+            StandardUsernamePasswordCredentials credentials = CredentialsHelper.lookupSystemCredentials(credentialsId, url != null ? url.toExternalForm() : null);
```

Comment:
could we move url null check inside `lookupSystemCredentials`?

Reply:
Good catch!